Mathlib/AlgebraicGeometry/AffineScheme.lean

```diff
@@ -677,6 +685,55 @@ lemma Scheme.eq_zeroLocus_of_isClosed_of_isAffine (X : Scheme.{u}) [IsAffine X]
 
 end ZeroLocus
 
+section Factorization
+
+/-- `f : X ⟶ Spec A` factors via some `Spec B` as `f = g ≫ h` such that `Γ(g)` is
+injective and `Γ(h)` is surjective. -/
+lemma exists_factorization_of_spec {X : Scheme.{u}} {A : CommRingCat} (f : X ⟶ Spec A) :
```

Comment:
I think this deserves definitions instead of exists statements. (`A` is the scheme theoretic image, and we might want to glue them later)

Reply:
I finally got to this and turned the exists statements into definitions.